routes/api/contacts.js

```diff
@@ -1,25 +1,124 @@
-const express = require('express')
+const express = require('express');
+const router = express.Router();
+const Joi = require('joi');
 
-const router = express.Router()
+const {
+  listContacts,
+  getContactById,
+  removeContact,
+  addContact,
+  updateContact,
+  updateStatusContact,
+} = require('../../models/contacts');
 
 router.get('/', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+  try {
+    const contacts = await listContacts();
+    res.status(200).json(contacts);
+  } catch (error) {
+    console.log(error);
+    next(error);
+  }
+});
 
-router.get('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+router.get('/:id', async (req, res, next) => {
+  const contactId = req.params.id;
+  try {
+    const contact = await getContactById(contactId);
+    res.status(200).json(contact);
+  } catch (error) {
+    res.status(404).json({ message: 'Not found' });
+    console.log(error);
+    next(error);
+  }
+});
 
 router.post('/', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+  const body = req.body;
 
-router.delete('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+  const schema = Joi.object().keys({
+    name: Joi.string().required(),
+    email: Joi.string()
+      .email()
+      .required(),
+    phone: Joi.string().required(),
+    favorite: Joi.boolean(),
+  });
+  const { error } = schema.validate(body);
+  if (error)
+    return res.status(400).json({ message: `${error.details[0].message}` });
 
-router.put('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+  try {
+    const contact = await addContact(body);
+    res.status(201).send(contact);
+  } catch (error) {
+    console.log(error);
+    next(error);
+  }
+});
 
-module.exports = router
+router.delete('/:id', async (req, res, next) => {
+  const contactId = req.params.id;
+  try {
+    const contact = await getContactById(contactId);
+    if (!contact) return res.status(404).json({ message: 'Not found' });
```

Comment:
Brak kontaktu tutaj wyrzuci błąd i odpali się kod z catch. Najlepiej było by zrobić try catch wewnątrz funkcji getContactById i tam jak wyrzuci błąd zwracać null i wtedy możesz tak to sprawdzać. Tylko wtedy musisz poprawić w innych miejscach gdzie używasz getContactById

Reply:
Mogę zmienić, ale sprawdziłem i jeżeli nie ma kontaktu to jest null(dodany console.log) i reszta działa poprawnie tzn. status 404 i message not found.
![delete bad id 2](https://user-images.githubusercontent.com/107140998/233685919-01e93387-32b2-4bc7-b3a8-46e9bc785a61.jpg)
![delete bad id](https://user-images.githubusercontent.com/107140998/233685922-1e90dec6-e2da-42d3-94ce-09b0cfc89730.jpg)

Reply:
Tak, o taki efekt chodziło by zwracało null'a i wtedy if (!contact) zadziała, wyrzuci 404 a catch będzie tylko dla jakichś wyjątków typu problem z bazą danych. Zastosował bym też to do poprzedniego komentarza oraz PUT'a i PATCH'a